ingestion-beam/src/main/java/com/mozilla/telemetry/Sink.java

```diff
@@ -67,4 +77,60 @@ static void registerOptions() {
     PipelineOptionsFactory.register(DecoderOptions.class);
   }
 
+  static class FlexTemplateMetadata {
+
+    public static void main(String[] args) {
+      run(SinkOptions.class);
+    }
+
+    static void run(Class<? extends PipelineOptions> optionsClass) {
+      final ArrayNode parameters = Json.createArrayNode();
+      for (Method method : optionsClass.getMethods()) {
+        if (PipelineOptions.class.equals(method.getDeclaringClass())
+            || !method.getName().startsWith("get")) {
+          // skip methods from the base class and non-getter methods
+          continue;
+        }
+        final String property = method.getName().substring(3);
+        final String name = property.substring(0, 1).toLowerCase() + property.substring(1);
+        final String label = property.replaceAll("([A-Z])", " $1").trim().replaceFirst("Bq", "BQ");
+        final boolean isOptional = method.getAnnotation(Required.class) == null;
+        final String description = method.getAnnotation(Description.class).value();
+
+        final String regex;
+        final String defaultValue;
+        Class<?> returnType = method.getReturnType();
+        if (returnType.isEnum()) {
+          regex = Arrays.asList((Enum[]) returnType.getEnumConstants()).stream().map(Enum::name)
+              .collect(Collectors.joining("|"));
+          final Default.Enum defaultAnnotation = method.getAnnotation(Default.Enum.class);
+          defaultValue = defaultAnnotation == null ? null : defaultAnnotation.value();
+        } else if (int.class.equals(returnType) || Integer.class.equals(returnType)) {
+          regex = "[0-9]+";
+          final Default.Integer defaultAnnotation = method.getAnnotation(Default.Integer.class);
+          defaultValue = defaultAnnotation == null ? null
+              : Integer.toString(defaultAnnotation.value());
+        } else if (long.class.equals(returnType) || Long.class.equals(returnType)) {
+          regex = "[0-9]+";
+          final Default.Long defaultAnnotation = method.getAnnotation(Default.Long.class);
+          defaultValue = defaultAnnotation == null ? null
+              : Long.toString(defaultAnnotation.value());
+        } else {
+          regex = ".*";
```

Comment:
parameters that use `ValueProvider` get this regex instead of checking their subtype, but I figure that's fine because #1534 includes removing `ValueProvider` stuff.